File: backend/src/shared/contracts/Shared.Contracts/RecoveryViaKeysSetRequest.cs
namespace Shared.Contracts
{
    public sealed record RecoveryViaKeysSetRequest(
        string Login,
        // Srp
        string EncryptedVerifier,
        string SrpSalt,
        int SrpVersion,
        string EncryptedVerifierWrapKey,
        int KeyWrapVersion,
        string AsymmetricKeyId,
        // Dek
        string EncryptedDek,
        string DekSalt, 
        int CryptoVersion,
        // Recovery Keys
        List<RecoveryKeyRequestData> RecoveryKeys);

        public record RecoveryKeyRequestData(string EncryptedValue, int CryptoVersion);
}

File: web/nexus-ui/src/features/auth/recovery-password-keys/components/set-password/models/recovery-via-keys-set.request.ts
export interface RecoveryViaKeysSetRequest {
    login: string;
    // Srp
    encryptedVerifier: string;
    srpSalt: string; 
    srpVersion: number;
    newCryptoVersion: number;
    encryptedVerifierWrapKey: string; 
    keyWrapVersion: number; 
    asymmetricKeyId: string;
    // Dek
    encryptedDek: string;
    dekSalt: string;
    cryptoVersion: number; 
    // Recovery Keys
    recoveryKeys: Array<RecoveryKeyRequestData>;
}

export interface RecoveryKeyRequestData {
    encryptedValue: string;
    cryptoVersion: number;
}